### ISSUE TITLE
[BUG][iOS] Crash when screen tapped quickly

### ISSUE THREAD

**Issue by mcezzare** [microsoft/clarity-apps#72]
**Describe the Bug**
App is crashing with clarity version (2.1.2)

**Repro Steps**
The crash happens when the user taps a control on the screen too many times quickly

**Expected Behavior**
Normal usage, don't crash without the pod

**Screenshots**
Can't upload image 

Fatal Exception: NSInvalidArgumentException
-[__SwiftValue doubleValue]: unrecognized selector sent to instance 0x300ac6d20

````
 Fatal Exception: NSInvalidArgumentException
0  CoreFoundation                 0x83f20 __exceptionPreprocess
1  libobjc.A.dylib                0x172b8 objc_exception_throw
2  CoreFoundation                 0x18d480 +[NSObject(NSObject) _copyDescription]
3  CoreFoundation                 0x20fb4 ___forwarding___
4  CoreFoundation                 0x208d0 _CF_forwarding_prep_0
5  QuartzCore                     0xb4264 -[NSNumber(CAAnimatableValue) CA_interpolateValue:byFraction:]
6  QuartzCore                     0xb0b50 -[CABasicAnimation applyForTime:presentationObject:modelObject:]
7  QuartzCore                     0xc3878 CA::Layer::presentation_layer(CA::Transaction*)
8  Clarity                        0xc9524 (Ausente UUID 0370cc1d7770379aa5a1d8eb736bd176)
9  Clarity                        0xc9e3c (Ausente UUID 0370cc1d7770379aa5a1d8eb736bd176)
10 Clarity                        0xc9e3c (Ausente UUID 0370cc1d7770379aa5a1d8eb736bd176)
11 Clarity                        0xc9e3c (Ausente UUID 0370cc1d7770379aa5a1d8eb736bd176)
12 Clarity                        0xc9e3c (Ausente UUID 0370cc1d7770379aa5a1d8eb736bd176)
13 Clarity                        0xc9e3c (Ausente UUID 0370cc1d7770379aa5a1d8eb736bd176)
14 Clarity                        0xc9e3c (Ausente UUID 0370cc1d7770379aa5a1d8eb736bd176)
15 Clarity                        0xc9e3c (Ausente UUID 0370cc1d7770379aa5a1d8eb736bd176)
16 Clarity                        0xc9e3c (Ausente UUID 0370cc1d7770379aa5a1d8eb736bd176)
17 Clarity                        0xc9e3c (Ausente UUID 0370cc1d7770379aa5a1d8eb736bd176)
18 Clarity                        0xc9e3c (Ausente UUID 0370cc1d7770379aa5a1d8eb736bd176)
19 Clarity                        0xc9e3c (Ausente UUID 0370cc1d7770379aa5a1d8eb736bd176)
20 Clarity                        0xc9e3c (Ausente UUID 0370cc1d7770379aa5a1d8eb736bd176)
21 Clarity                        0xc9e3c (Ausente UUID 0370cc1d7770379aa5a1d8eb736bd176)
22 Clarity                        0xc8ee4 (Ausente UUID 0370cc1d7770379aa5a1d8eb736bd176)
23 Clarity                        0xc8b24 (Ausente UUID 0370cc1d7770379aa5a1d8eb736bd176)
24 Clarity                        0xf44d8 (Ausente UUID 0370cc1d7770379aa5a1d8eb736bd176)
25 Clarity                        0xf3f98 (Ausente UUID 0370cc1d7770379aa5a1d8eb736bd176)
26 Clarity                        0xa1b18 (Ausente UUID 0370cc1d7770379aa5a1d8eb736bd176)
27 libdispatch.dylib              0x3dd4 _dispatch_client_callout
28 libdispatch.dylib              0x72d8 _dispatch_continuation_pop
29 libdispatch.dylib              0x1b1c8 _dispatch_source_latch_and_call
30 libdispatch.dylib              0x19d8c _dispatch_source_invoke
31 libdispatch.dylib              0x124b4 _dispatch_main_queue_drain
32 libdispatch.dylib              0x121b8 _dispatch_main_queue_callback_4CF
33 CoreFoundation                 0x56710 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__
34 CoreFoundation                 0x53914 __CFRunLoopRun
35 CoreFoundation                 0x52cd8 CFRunLoopRunSpecific
36 GraphicsServices               0x11a8 GSEventRunModal
37 UIKitCore                      0x40aae8 -[UIApplication _run]
38 UIKitCore                      0x4bed98 UIApplicationMain
39 UIKitCore                      0x638504 keypath_get_selector_hoverStyle
40 APPNAME_HIDDEN     0x1b2fa4 main (AppDelegate.swift)
41 ???                            0x1c0707154 (Ausente)
````

And strange that line of clarity is 
Clarity                        0xc9524 **(Ausente UUID 0370cc1d7770379aa5a1d8eb736bd176)**

**Project Info**
Pod version 2.12, its recording fine a lot of sessions, when the crash happens the session recorded is not uploaded

- Project ID (share with clarity-apps-support@microsoft.com while mentioning the issue number in the subject)

**Platform/Framework Info**

- Device: Many iPhone models, Iphone 8, XR, 11, 13, 14, 11
- OS: ios  15.8.3, 16.7.10, 17.6.1 
- Application Framework [Native UIKit-based]
- Clarity Version [2.1.2]

**Additional Context**
This crash is happening only when the user taps a control on the screen too many times quickly, example password textfields or back on navigation too fast.

**Comment by amralaa-MSFT** [comment]
@mcezzare: Thanks for reporting this crash! To help us investigate further, we'd appreciate some additional details:

- Project Information: Please share your project ID with us through email at clarity-apps-support@microsoft.com. Mention the issue number (#72) in the subject line for easy reference.
- Crash Details:
    - Consistency: Does the crash happen every time the screen is tapped quickly, or is it sporadic?
    - UI Correlation: Is the crash related to any specific UI view classes or types? (e.g., only password fields, navigation buttons) We haven't been able to reproduce this in other apps.
    - Password Field Implementation: Can you describe how you implemented the password field? Which UI components are used? (e.g., UITextField, SecureTextField)
    - Code Snippet (if possible): If possible, could you share a minimal code snippet of a UIViewController that triggers the crash? This would be incredibly helpful for debugging.

We appreciate your cooperation in helping us diagnose this issue. Once we have this additional information, we'll be able to investigate further and work towards a solution.

**Comment by SherifShokry123** [comment]
@amralaa-MSFT We're experiencing the same crash in production. 
I can consistently reproduce it by clicking on a video in the chat screen to open it in full screen. 
The stack trace we’re seeing matches the one shared above.

**Comment by ibradwan** [comment]
Hi,

This issue should be resolved in iOS SDK version 2.2.2. Please upgrade and let us know if you have any feedback.
